### PR TITLE
Fix models belongs_to records

### PR DESCRIPTION
**Pull request by rderik**
All models were checked for fields that have the relationship belongs to
and verify if they should be optional or not_null, this commit includes
migrations to fix the models affected.

### CODE REVIEW

**Comment by rderik** [comment]
This branch contains all the changes made to kmaps_engine after the revision on data models and the verification of the views related to those models. The validation was triggered after the upgrade to rails 5.1 made belongs_to association to be by default false, that had the effect of breaking some of the code this branch addresses that problem.

File: app/models/feature_geo_code.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@ class FeatureGeoCode < ActiveRecord::Base
   belongs_to :feature
   belongs_to :geo_code_type
   has_many :imports, :as => 'item', :dependent => :destroy
+
+  validates :geo_code_type_id, :feature_id, :geo_code_value, presence: true
```
Comment (amontano) [inline app/models/feature_geo_code.rb]: only leave geo_code_value in "validates". others are getting validated by belongs_to

Comment (rderik) [reply app/models/feature_geo_code.rb]: Fixed

File: app/models/citation.rb [hunk 2]
```diff
@@ -33,6 +33,7 @@ class Citation < ActiveRecord::Base
   # Validation
   #
   #
+  validates :info_source_id, :citable_id, :citable_type, :info_source_type, presence: true
   validates :info_source_id, presence: true
```
[the remark this answers is not in the document]
Comment (rderik) [reply app/models/citation.rb]: Fixed

**Pull request merged**